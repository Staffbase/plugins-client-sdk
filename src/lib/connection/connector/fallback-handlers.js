/**
 * Fallbacks for all sdk commands
 */
let log = require('loglevel');
let userAgent = navigator.userAgent || navigator.vendor || window.opera || '';
/**
 * Get the current Staffbase app version
 *
 * @return {String} version
 */
export function getVersion() {
  log.debug('fallback/getVersion');
  return '3.4';
}

/**
 * Are we running in a native app
 *
 * works only for ios
 * @return {Boolean}
 */
export function isNative() {
  log.debug('fallback/isNative');
  let safari = /safari/i.test(userAgent);
  return !safari && isIos();
}

/**
 * Are we running on a mobile device
 *
 * @return {Boolean}
 */
export function isMobile() {
  log.debug('fallback/isMobile');
  return /Android|webOS|iPhone|iPad|BlackBerry|BB10|IEMobile|Opera Mini/i.test(userAgent);
}

/**
 * Are we running on a desktop device
 *
 * @return {Boolean}
 */
export function isDesktop() {
  log.debug('fallback/isDesktop');
  return /Win|Mac|Linux/i.test(userAgent);
}

/**
 * Are we running on android
 *
 * @return {Boolean}
 */
export function isAndroid() {
  log.debug('fallback/isAndroid');
  return /Android/i.test(userAgent);
}

/**
 * Are we running on ios
 *
 * @return {Boolean}
 */
export function isIos() {
  log.debug('fallback/isIos');
  return /iPhone|iPad|iPod/i.test(userAgent);
}

/**
 * Open an external link
 *
 * @param {String} url address
 */
export function openLink(url) {
  log.debug('fallback/openLink');
  window.open(url, '_blank');
}

/**
 * Open an external link
 *
 * @param {String} url address
 */
export function nativeUpload(url) {
  log.debug('fallback/nativeUpload');
  log.warn('Native upload is not possible in fallback mode.');
  // nothing we can do here
  return;
}

/**
 * Get extensive locale information.
 *
 * @return {Object} containing various language informations
 */
export function langInfos() {
  log.debug('fallback/langInfos');
  return {
    contentLanguages: ['en_EN', 'de_DE']
  };
}
