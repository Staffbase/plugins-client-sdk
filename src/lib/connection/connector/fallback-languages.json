{
    "ar": {
        "key": "ar",
        "locale": "ar_AE",
        "name": "العربية",
        "localizedName": "Arabic"
    },
    "bg": {
        "key": "bg",
        "locale": "bg_BG",
        "name": "български",
        "localizedName": "Bulgarian"
    },
    "bn": {
        "key": "bn",
        "locale": "bn_BD",
        "name": "বাংলা",
        "localizedName": "Bengali"
    },
    "cs": {
        "key": "cs",
        "locale": "cs_CZ",
        "name": "Čeština",
        "localizedName": "Czech"
    },
    "da": {
        "key": "da",
        "locale": "da_DK",
        "name": "Dansk",
        "localizedName": "Danish"
    },
    "de": {
        "key": "de",
        "locale": "de_DE",
        "name": "Deutsch",
        "localizedName": "German"
    },
    "el": {
        "key": "el",
        "locale": "el_GR",
        "name": "Ελληνικά",
        "localizedName": "Greek"
    },
    "en": {
        "key": "en",
        "locale": "en_US",
        "name": "English",
        "localizedName": "English"
    },
    "es": {
        "key": "es",
        "locale": "es_ES",
        "name": "Español",
        "localizedName": "Spanish"
    },
    "et": {
        "key": "et",
        "locale": "et_EE",
        "name": "Eesti keel",
        "localizedName": "Estonian"
    },
    "fi": {
        "key": "fi",
        "locale": "fi_FI",
        "name": "Suomi",
        "localizedName": "Finnish"
    },
    "fr": {
        "key": "fr",
        "locale": "fr_FR",
        "name": "Français",
        "localizedName": "French"
    },
    "he": {
        "key": "he",
        "locale": "he_IL",
        "name": "עִבְרִית",
        "localizedName": "Hebrew"
    },
    "hi": {
        "key": "hi",
        "locale": "hi_IN",
        "name": "िन्दी",
        "localizedName": "Hindi"
    },
    "hr": {
        "key": "hr",
        "locale": "hr_HR",
        "name": "Hrvatski",
        "localizedName": "Croatian"
    },
    "hu": {
        "key": "hu",
        "locale": "hu_HU",
        "name": "Magyar",
        "localizedName": "Hungarian"
    },
    "is": {
        "key": "is",
        "locale": "is_IS",
        "name": "Íslenska",
        "localizedName": "Icelandic"
    },
    "it": {
        "key": "it",
        "locale": "it_IT",
        "name": "Italiano",
        "localizedName": "Italian"
    },
    "ja": {
        "key": "ja",
        "locale": "ja_JP",
        "name": "日本語",
        "localizedName": "Japanese"
    },
    "ko": {
        "key": "ko",
        "locale": "ko_KR",
        "name": "한국어",
        "localizedName": "Korean"
    },
    "lt": {
        "key": "lt",
        "locale": "lt_LT",
        "name": "Lietuvių",
        "localizedName": "Lithuanian"
    },
    "lv": {
        "key": "lv",
        "locale": "lv_LV",
        "name": "Latviešu",
        "localizedName": "Latvian"
    },
    "mk": {
        "key": "mk",
        "locale": "mk_MK",
        "name": "Македонски",
        "localizedName": "Macedonian"
    },
    "ms": {
        "key": "ms",
        "locale": "ms_MY",
        "name": "Bahasa Melayu",
        "localizedName": "Malay"
    },
    "nl": {
        "key": "nl",
        "locale": "nl_NL",
        "name": "Nederlands",
        "localizedName": "Dutch"
    },
    "no": {
        "key": "no",
        "locale": "no_NO",
        "name": "Norsk",
        "localizedName": "Norwegian"
    },
    "pl": {
        "key": "pl",
        "locale": "pl_PL",
        "name": "Polski",
        "localizedName": "Polish"
    },
    "pt": {
        "key": "pt",
        "locale": "pt_BR",
        "name": "Português",
        "localizedName": "Portuguese"
    },
    "ro": {
        "key": "ro",
        "locale": "ro_RO",
        "name": "Română",
        "localizedName": "Romanian"
    },
    "ru": {
        "key": "ru",
        "locale": "ru_RU",
        "name": "Русский",
        "localizedName": "Russian"
    },
    "sk": {
        "key": "sk",
        "locale": "sk_SK",
        "name": "Slovenský",
        "localizedName": "Slovak"
    },
    "sl": {
        "key": "sl",
        "locale": "sl_SI",
        "name": "Slovenščina",
        "localizedName": "Slovenian"
    },
    "sr": {
        "key": "sr",
        "locale": "sr_RS",
        "name": "Српски",
        "localizedName": "Serbian"
    },
    "sv": {
        "key": "sv",
        "locale": "sv_SE",
        "name": "Svenska",
        "localizedName": "Swedish"
    },
    "th": {
        "key": "th",
        "locale": "th_TH",
        "name": "ภาษาไทย",
        "localizedName": "Thai"
    },
    "tr": {
        "key": "tr",
        "locale": "tr_TR",
        "name": "Türkçe",
        "localizedName": "Turkish"
    },
    "uk": {
        "key": "uk",
        "locale": "uk_UA",
        "name": "Українська",
        "localizedName": "Ukrainian"
    },
    "vi": {
        "key": "vi",
        "locale": "vi_VN",
        "name": "Tiếng Việt",
        "localizedName": "Vietnamese"
    },
    "zh": {
        "key": "zh",
        "locale": "zh_CN",
        "name": "简体中文",
        "localizedName": "Chinese"
    }
}