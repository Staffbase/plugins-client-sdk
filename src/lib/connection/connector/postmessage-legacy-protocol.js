/**
 * Postmessage legacy protocol (3.5)
 */

export default {
  init: 'pluginLoaded', // send this if ready to communicate
  platformInfo: 'platformInfo' // receive this after pluginLoaded was sent
};
