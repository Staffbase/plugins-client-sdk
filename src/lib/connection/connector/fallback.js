import { commands as action } from '../commands';
import * as fallbacks from './fallback-handlers';
let log = require('loglevel');

let connection = null;
const fallbackKickIn = 500;

/**
 * Fake connection to the Staffbase App after a period of time.
 *
 * The fallback message handler will signal an established connection
 * after the time specified in fallbackKickIn runs out.
 * @return {Promise<function>} An appropriate send function
 */
export default function connect() {
  if (connection) {
    throw new Error('Connect called twice.');
  }

  connection = new Promise((resolve, reject) => {
    setTimeout(function() {
      log.info('fallback/connect succeeded');
      resolve(sendMessage);
    }, fallbackKickIn);
  });

  return connection;
}

/**
 * Disconnect from the Staffbase App
 *
 * Only usefull for tests.
 */
export function disconnect() {
  connection = null;
}

/**
 * Send a SDK command to the Staffbase App.
 *
 * Maps SDK commands to fallback handlers.
 * @param {String} cmd an SDK command
 * @param {any} payload for the command
 * @return {Promise<any>} which awaits the response of the Staffbase App
 * @throws {Error} on commands not supported by protocol
 */
export async function sendMessage(cmd, ...payload) {
  log.info('fallback/sendMessage ' + cmd);
  log.debug('fallback/sendMessage/payload ' + JSON.stringify(payload));

  switch (cmd) {
    case action.version:
      return fallbacks.getVersion();
    case action.native:
      return fallbacks.isNative();
    case action.mobile:
      return fallbacks.isMobile();
    case action.ios:
      return fallbacks.isIos();
    case action.android:
      return fallbacks.isAndroid();
    case action.openLink:
      return fallbacks.openLink.apply(null, payload);
    case action.nativeUpload:
      return fallbacks.nativeUpload();
    case action.langInfos:
      return fallbacks.langInfos();
    case action.branchDefaultLang:
      return fallbacks.getBranchDefaultLanguage();
    case action.prefContentLang:
      return fallbacks.getPreferredContentLocale.apply(null, payload);
    default:
      // should actualy never ever happen
      throw new Error('Command ' + cmd + ' not supported by driver');
  }
}
