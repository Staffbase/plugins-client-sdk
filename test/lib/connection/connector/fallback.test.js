import connect from '../../../../src/lib/connection/connector/fallback';
import { disconnect } from '../../../../src/lib/connection/connector/fallback';
import command from '../../../../src/lib/connection/commands';

let chai = require('chai');
let expect = chai.expect;

describe('connector/fallback', function() {
  describe('connect', function() {
    afterEach(function() {
      disconnect();
    });

    it('should be a function', () => {
      expect(connect).to.be.a('function');
    });

    it('should yield a Promise', () => {
      let res = connect();
      expect(res).to.be.a('promise');
    });

    it('should provide a function', async () => {
      let fn = await connect();
      expect(fn).to.be.a('function');
    });

    it('should resolve after 500 ms', async () => {
      let start = new Date().getTime();
      return connect().then(() => {
        let end = new Date().getTime();
        expect(end - start).to.be.at.least(499);
      });
    });

    it('should resolve before 600 ms', async () => {
      let start = new Date().getTime();
      return connect().then(() => {
        let end = new Date().getTime();
        expect(end - start).to.be.at.most(600);
      });
    });

    describe('send function', async function() {
      it('should reject on unknown commands', async () => {
        let sendFn = await connect();
        return expect(sendFn('unknown-asdf-command')).to.be.rejected;
      });

      describe('accepts all comands', async function() {
        // mock window open
        window.open = function() {};

        let commandData = {
          prefContentLang: ['de_DE', 'en_US']
        };

        for (let cmd in command) {
          if (command.hasOwnProperty(cmd)) {
            it('command.' + cmd, async () => {
              let sendFn = await connect();
              return expect(sendFn(command[cmd], commandData[cmd])).to.be.fulfilled;
            });
          }
        }
      });
    });
  });
});
