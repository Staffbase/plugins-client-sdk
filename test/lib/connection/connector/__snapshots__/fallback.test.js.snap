// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connector/fallback connect send function accepts all comands command.android 1`] = `false`;

exports[`connector/fallback connect send function accepts all comands command.branchDefaultLang 1`] = `
Object {
  "key": "en",
  "locale": "en_US",
  "localizedName": "English",
  "name": "English",
}
`;

exports[`connector/fallback connect send function accepts all comands command.ios 1`] = `false`;

exports[`connector/fallback connect send function accepts all comands command.langInfos 1`] = `
Object {
  "branchDefaultLanguage": Object {
    "key": "en",
    "locale": "en_US",
    "localizedName": "English",
    "name": "English",
  },
  "branchLanguages": Object {
    "af": Object {
      "key": "af",
      "locale": "af_ZA",
      "localizedName": "Afrikaans",
      "name": "Afrikaans",
    },
    "am": Object {
      "key": "am",
      "locale": "am_ET",
      "localizedName": "Amharic",
      "name": "አማርኛ",
    },
    "ar": Object {
      "direction": "rtl",
      "key": "ar",
      "locale": "ar_SA",
      "localizedName": "Arabic (Saudi Arabia)",
      "name": "العربية (المملكة العربية السعودية)",
    },
    "arAE": Object {
      "direction": "rtl",
      "key": "arAE",
      "locale": "ar_AE",
      "localizedName": "Arabic (United Arab Emirates)",
      "name": "العربية (الإمارات العربية المتحدة)",
    },
    "arEG": Object {
      "direction": "rtl",
      "key": "arEG",
      "locale": "ar_EG",
      "localizedName": "Arabic (Egypt)",
      "name": "العربية (مصر)",
    },
    "bg": Object {
      "key": "bg",
      "locale": "bg_BG",
      "localizedName": "Bulgarian",
      "name": "български",
    },
    "bn": Object {
      "key": "bn",
      "locale": "bn_BD",
      "localizedName": "Bengali",
      "name": "বাংলা",
    },
    "bs": Object {
      "key": "bs",
      "locale": "bs_BA",
      "localizedName": "Bosnian",
      "name": "Bosanski",
    },
    "cs": Object {
      "key": "cs",
      "locale": "cs_CZ",
      "localizedName": "Czech",
      "name": "Čeština",
    },
    "cy": Object {
      "key": "cy",
      "locale": "cy_GB",
      "localizedName": "Welsh",
      "name": "Cymraeg",
    },
    "da": Object {
      "key": "da",
      "locale": "da_DK",
      "localizedName": "Danish",
      "name": "Dansk",
    },
    "de": Object {
      "key": "de",
      "locale": "de_DE",
      "localizedName": "German",
      "name": "Deutsch",
    },
    "el": Object {
      "key": "el",
      "locale": "el_GR",
      "localizedName": "Greek",
      "name": "Ελληνικά",
    },
    "en": Object {
      "key": "en",
      "locale": "en_US",
      "localizedName": "English",
      "name": "English",
    },
    "es": Object {
      "key": "es",
      "locale": "es_ES",
      "localizedName": "Spanish (Spain)",
      "name": "Español (España)",
    },
    "esMX": Object {
      "key": "esMX",
      "locale": "es_MX",
      "localizedName": "Spanish (Mexico)",
      "name": "Español (México)",
    },
    "et": Object {
      "key": "et",
      "locale": "et_EE",
      "localizedName": "Estonian",
      "name": "Eesti keel",
    },
    "fi": Object {
      "key": "fi",
      "locale": "fi_FI",
      "localizedName": "Finnish",
      "name": "Suomi",
    },
    "fr": Object {
      "key": "fr",
      "locale": "fr_FR",
      "localizedName": "French (France)",
      "name": "Français (France)",
    },
    "frCA": Object {
      "key": "frCA",
      "locale": "fr_CA",
      "localizedName": "French (Canada)",
      "name": "Français (Canada)",
    },
    "he": Object {
      "direction": "rtl",
      "key": "he",
      "locale": "he_IL",
      "localizedName": "Hebrew",
      "name": "עִבְרִית",
    },
    "hi": Object {
      "key": "hi",
      "locale": "hi_IN",
      "localizedName": "Hindi",
      "name": "िन्दी",
    },
    "hk": Object {
      "key": "hk",
      "locale": "zh_HK",
      "localizedName": "Chinese traditional",
      "name": "繁體中文",
    },
    "hr": Object {
      "key": "hr",
      "locale": "hr_HR",
      "localizedName": "Croatian",
      "name": "Hrvatski",
    },
    "ht": Object {
      "key": "ht",
      "locale": "ht_HT",
      "localizedName": "Haitian Creole",
      "name": "kreyòl ayisyen",
    },
    "hu": Object {
      "key": "hu",
      "locale": "hu_HU",
      "localizedName": "Hungarian",
      "name": "Magyar",
    },
    "in": Object {
      "key": "in",
      "locale": "in_ID",
      "localizedName": "Indonesian",
      "name": "Bahasa Indonesia",
    },
    "is": Object {
      "key": "is",
      "locale": "is_IS",
      "localizedName": "Icelandic",
      "name": "Íslenska",
    },
    "it": Object {
      "key": "it",
      "locale": "it_IT",
      "localizedName": "Italian",
      "name": "Italiano",
    },
    "ja": Object {
      "key": "ja",
      "locale": "ja_JP",
      "localizedName": "Japanese",
      "name": "日本語",
    },
    "kl": Object {
      "key": "kl",
      "locale": "kl_GL",
      "localizedName": "Greenlandic",
      "name": "Kalaallisut",
    },
    "km": Object {
      "key": "km",
      "locale": "km_KH",
      "localizedName": "Khmer",
      "name": "ខ្មែរ",
    },
    "ko": Object {
      "key": "ko",
      "locale": "ko_KR",
      "localizedName": "Korean",
      "name": "한국어",
    },
    "lo": Object {
      "key": "lo",
      "locale": "lo_LA",
      "localizedName": "Lao",
      "name": "ລາວ",
    },
    "lt": Object {
      "key": "lt",
      "locale": "lt_LT",
      "localizedName": "Lithuanian",
      "name": "Lietuvių",
    },
    "lv": Object {
      "key": "lv",
      "locale": "lv_LV",
      "localizedName": "Latvian",
      "name": "Latviešu",
    },
    "mg": Object {
      "key": "mg",
      "locale": "mg_MG",
      "localizedName": "Malagasy",
      "name": "Malagasy",
    },
    "mh": Object {
      "key": "mh",
      "locale": "mh_MH",
      "localizedName": "Marshallese",
      "name": "Kajin M̧ajeļ",
    },
    "mi": Object {
      "key": "mi",
      "locale": "mi_NZ",
      "localizedName": "Māori",
      "name": "Māori",
    },
    "mk": Object {
      "key": "mk",
      "locale": "mk_MK",
      "localizedName": "Macedonian",
      "name": "Македонски",
    },
    "mn": Object {
      "key": "mn",
      "locale": "mn_MN",
      "localizedName": "Mongolian",
      "name": "монгол",
    },
    "mr": Object {
      "key": "mr",
      "locale": "mr_IN",
      "localizedName": "Marathi",
      "name": "मराठी",
    },
    "ms": Object {
      "key": "ms",
      "locale": "ms_MY",
      "localizedName": "Malay",
      "name": "Bahasa Melayu",
    },
    "my": Object {
      "key": "my",
      "locale": "my_MM",
      "localizedName": "Burmese",
      "name": "Burmese",
    },
    "ne": Object {
      "key": "ne",
      "locale": "ne_NP",
      "localizedName": "Nepali",
      "name": "नेपाली",
    },
    "nl": Object {
      "key": "nl",
      "locale": "nl_NL",
      "localizedName": "Dutch",
      "name": "Nederlands",
    },
    "nlBE": Object {
      "key": "nlBE",
      "locale": "nl_BE",
      "localizedName": "Flemish",
      "name": "Vlaemsch",
    },
    "no": Object {
      "key": "no",
      "locale": "no_NO",
      "localizedName": "Norwegian",
      "name": "Norsk",
    },
    "pl": Object {
      "key": "pl",
      "locale": "pl_PL",
      "localizedName": "Polish",
      "name": "Polski",
    },
    "pt": Object {
      "key": "pt",
      "locale": "pt_PT",
      "localizedName": "Portuguese (Portugal)",
      "name": "Português (Portugal)",
    },
    "ptBR": Object {
      "key": "ptBR",
      "locale": "pt_BR",
      "localizedName": "Portuguese (Brazil)",
      "name": "Português (Brasil)",
    },
    "ro": Object {
      "key": "ro",
      "locale": "ro_RO",
      "localizedName": "Romanian",
      "name": "Română",
    },
    "ru": Object {
      "key": "ru",
      "locale": "ru_RU",
      "localizedName": "Russian",
      "name": "Русский",
    },
    "si": Object {
      "key": "si",
      "locale": "si_LK",
      "localizedName": "Sinhala",
      "name": "සිංහල",
    },
    "sk": Object {
      "key": "sk",
      "locale": "sk_SK",
      "localizedName": "Slovak",
      "name": "Slovenský",
    },
    "sl": Object {
      "key": "sl",
      "locale": "sl_SI",
      "localizedName": "Slovenian",
      "name": "Slovenščina",
    },
    "so": Object {
      "key": "so",
      "locale": "so_SO",
      "localizedName": "Somali",
      "name": "Af-Soomaali",
    },
    "sq": Object {
      "key": "sq",
      "locale": "sq_AL",
      "localizedName": "Albanian",
      "name": "Shqip",
    },
    "sr": Object {
      "key": "sr",
      "locale": "sr_RS",
      "localizedName": "Serbian",
      "name": "Српски",
    },
    "sv": Object {
      "key": "sv",
      "locale": "sv_SE",
      "localizedName": "Swedish",
      "name": "Svenska",
    },
    "sw": Object {
      "key": "sw",
      "locale": "sw_TZ",
      "localizedName": "Swahili",
      "name": "Swahili",
    },
    "ta": Object {
      "key": "ta",
      "locale": "ta_IN",
      "localizedName": "Tamil",
      "name": "தமிழ்",
    },
    "th": Object {
      "key": "th",
      "locale": "th_TH",
      "localizedName": "Thai",
      "name": "ภาษาไทย",
    },
    "tl": Object {
      "key": "tl",
      "locale": "tl_PH",
      "localizedName": "Filipino",
      "name": "Filipino",
    },
    "tr": Object {
      "key": "tr",
      "locale": "tr_TR",
      "localizedName": "Turkish",
      "name": "Türkçe",
    },
    "uk": Object {
      "key": "uk",
      "locale": "uk_UA",
      "localizedName": "Ukrainian",
      "name": "Українська",
    },
    "ur": Object {
      "direction": "rtl",
      "key": "ur",
      "locale": "ur_PK",
      "localizedName": "Urdu",
      "name": "اُردُو",
    },
    "vi": Object {
      "key": "vi",
      "locale": "vi_VN",
      "localizedName": "Vietnamese",
      "name": "Tiếng Việt",
    },
    "zh": Object {
      "key": "zh",
      "locale": "zh_CN",
      "localizedName": "Chinese",
      "name": "简体中文",
    },
    "zu": Object {
      "key": "zu",
      "locale": "zu_ZA",
      "localizedName": "Zulu",
      "name": "isiZulu",
    },
  },
  "contentLanguage": Object {
    "key": "en",
    "locale": "en_US",
    "localizedName": "English",
    "name": "English",
  },
  "contentLanguages": Object {
    "af": Object {
      "key": "af",
      "locale": "af_ZA",
      "localizedName": "Afrikaans",
      "name": "Afrikaans",
    },
    "am": Object {
      "key": "am",
      "locale": "am_ET",
      "localizedName": "Amharic",
      "name": "አማርኛ",
    },
    "ar": Object {
      "direction": "rtl",
      "key": "ar",
      "locale": "ar_SA",
      "localizedName": "Arabic (Saudi Arabia)",
      "name": "العربية (المملكة العربية السعودية)",
    },
    "arAE": Object {
      "direction": "rtl",
      "key": "arAE",
      "locale": "ar_AE",
      "localizedName": "Arabic (United Arab Emirates)",
      "name": "العربية (الإمارات العربية المتحدة)",
    },
    "arEG": Object {
      "direction": "rtl",
      "key": "arEG",
      "locale": "ar_EG",
      "localizedName": "Arabic (Egypt)",
      "name": "العربية (مصر)",
    },
    "bg": Object {
      "key": "bg",
      "locale": "bg_BG",
      "localizedName": "Bulgarian",
      "name": "български",
    },
    "bn": Object {
      "key": "bn",
      "locale": "bn_BD",
      "localizedName": "Bengali",
      "name": "বাংলা",
    },
    "bs": Object {
      "key": "bs",
      "locale": "bs_BA",
      "localizedName": "Bosnian",
      "name": "Bosanski",
    },
    "cs": Object {
      "key": "cs",
      "locale": "cs_CZ",
      "localizedName": "Czech",
      "name": "Čeština",
    },
    "cy": Object {
      "key": "cy",
      "locale": "cy_GB",
      "localizedName": "Welsh",
      "name": "Cymraeg",
    },
    "da": Object {
      "key": "da",
      "locale": "da_DK",
      "localizedName": "Danish",
      "name": "Dansk",
    },
    "de": Object {
      "key": "de",
      "locale": "de_DE",
      "localizedName": "German",
      "name": "Deutsch",
    },
    "el": Object {
      "key": "el",
      "locale": "el_GR",
      "localizedName": "Greek",
      "name": "Ελληνικά",
    },
    "en": Object {
      "key": "en",
      "locale": "en_US",
      "localizedName": "English",
      "name": "English",
    },
    "es": Object {
      "key": "es",
      "locale": "es_ES",
      "localizedName": "Spanish (Spain)",
      "name": "Español (España)",
    },
    "esMX": Object {
      "key": "esMX",
      "locale": "es_MX",
      "localizedName": "Spanish (Mexico)",
      "name": "Español (México)",
    },
    "et": Object {
      "key": "et",
      "locale": "et_EE",
      "localizedName": "Estonian",
      "name": "Eesti keel",
    },
    "fi": Object {
      "key": "fi",
      "locale": "fi_FI",
      "localizedName": "Finnish",
      "name": "Suomi",
    },
    "fr": Object {
      "key": "fr",
      "locale": "fr_FR",
      "localizedName": "French (France)",
      "name": "Français (France)",
    },
    "frCA": Object {
      "key": "frCA",
      "locale": "fr_CA",
      "localizedName": "French (Canada)",
      "name": "Français (Canada)",
    },
    "he": Object {
      "direction": "rtl",
      "key": "he",
      "locale": "he_IL",
      "localizedName": "Hebrew",
      "name": "עִבְרִית",
    },
    "hi": Object {
      "key": "hi",
      "locale": "hi_IN",
      "localizedName": "Hindi",
      "name": "िन्दी",
    },
    "hk": Object {
      "key": "hk",
      "locale": "zh_HK",
      "localizedName": "Chinese traditional",
      "name": "繁體中文",
    },
    "hr": Object {
      "key": "hr",
      "locale": "hr_HR",
      "localizedName": "Croatian",
      "name": "Hrvatski",
    },
    "ht": Object {
      "key": "ht",
      "locale": "ht_HT",
      "localizedName": "Haitian Creole",
      "name": "kreyòl ayisyen",
    },
    "hu": Object {
      "key": "hu",
      "locale": "hu_HU",
      "localizedName": "Hungarian",
      "name": "Magyar",
    },
    "in": Object {
      "key": "in",
      "locale": "in_ID",
      "localizedName": "Indonesian",
      "name": "Bahasa Indonesia",
    },
    "is": Object {
      "key": "is",
      "locale": "is_IS",
      "localizedName": "Icelandic",
      "name": "Íslenska",
    },
    "it": Object {
      "key": "it",
      "locale": "it_IT",
      "localizedName": "Italian",
      "name": "Italiano",
    },
    "ja": Object {
      "key": "ja",
      "locale": "ja_JP",
      "localizedName": "Japanese",
      "name": "日本語",
    },
    "kl": Object {
      "key": "kl",
      "locale": "kl_GL",
      "localizedName": "Greenlandic",
      "name": "Kalaallisut",
    },
    "km": Object {
      "key": "km",
      "locale": "km_KH",
      "localizedName": "Khmer",
      "name": "ខ្មែរ",
    },
    "ko": Object {
      "key": "ko",
      "locale": "ko_KR",
      "localizedName": "Korean",
      "name": "한국어",
    },
    "lo": Object {
      "key": "lo",
      "locale": "lo_LA",
      "localizedName": "Lao",
      "name": "ລາວ",
    },
    "lt": Object {
      "key": "lt",
      "locale": "lt_LT",
      "localizedName": "Lithuanian",
      "name": "Lietuvių",
    },
    "lv": Object {
      "key": "lv",
      "locale": "lv_LV",
      "localizedName": "Latvian",
      "name": "Latviešu",
    },
    "mg": Object {
      "key": "mg",
      "locale": "mg_MG",
      "localizedName": "Malagasy",
      "name": "Malagasy",
    },
    "mh": Object {
      "key": "mh",
      "locale": "mh_MH",
      "localizedName": "Marshallese",
      "name": "Kajin M̧ajeļ",
    },
    "mi": Object {
      "key": "mi",
      "locale": "mi_NZ",
      "localizedName": "Māori",
      "name": "Māori",
    },
    "mk": Object {
      "key": "mk",
      "locale": "mk_MK",
      "localizedName": "Macedonian",
      "name": "Македонски",
    },
    "mn": Object {
      "key": "mn",
      "locale": "mn_MN",
      "localizedName": "Mongolian",
      "name": "монгол",
    },
    "mr": Object {
      "key": "mr",
      "locale": "mr_IN",
      "localizedName": "Marathi",
      "name": "मराठी",
    },
    "ms": Object {
      "key": "ms",
      "locale": "ms_MY",
      "localizedName": "Malay",
      "name": "Bahasa Melayu",
    },
    "my": Object {
      "key": "my",
      "locale": "my_MM",
      "localizedName": "Burmese",
      "name": "Burmese",
    },
    "ne": Object {
      "key": "ne",
      "locale": "ne_NP",
      "localizedName": "Nepali",
      "name": "नेपाली",
    },
    "nl": Object {
      "key": "nl",
      "locale": "nl_NL",
      "localizedName": "Dutch",
      "name": "Nederlands",
    },
    "nlBE": Object {
      "key": "nlBE",
      "locale": "nl_BE",
      "localizedName": "Flemish",
      "name": "Vlaemsch",
    },
    "no": Object {
      "key": "no",
      "locale": "no_NO",
      "localizedName": "Norwegian",
      "name": "Norsk",
    },
    "pl": Object {
      "key": "pl",
      "locale": "pl_PL",
      "localizedName": "Polish",
      "name": "Polski",
    },
    "pt": Object {
      "key": "pt",
      "locale": "pt_PT",
      "localizedName": "Portuguese (Portugal)",
      "name": "Português (Portugal)",
    },
    "ptBR": Object {
      "key": "ptBR",
      "locale": "pt_BR",
      "localizedName": "Portuguese (Brazil)",
      "name": "Português (Brasil)",
    },
    "ro": Object {
      "key": "ro",
      "locale": "ro_RO",
      "localizedName": "Romanian",
      "name": "Română",
    },
    "ru": Object {
      "key": "ru",
      "locale": "ru_RU",
      "localizedName": "Russian",
      "name": "Русский",
    },
    "si": Object {
      "key": "si",
      "locale": "si_LK",
      "localizedName": "Sinhala",
      "name": "සිංහල",
    },
    "sk": Object {
      "key": "sk",
      "locale": "sk_SK",
      "localizedName": "Slovak",
      "name": "Slovenský",
    },
    "sl": Object {
      "key": "sl",
      "locale": "sl_SI",
      "localizedName": "Slovenian",
      "name": "Slovenščina",
    },
    "so": Object {
      "key": "so",
      "locale": "so_SO",
      "localizedName": "Somali",
      "name": "Af-Soomaali",
    },
    "sq": Object {
      "key": "sq",
      "locale": "sq_AL",
      "localizedName": "Albanian",
      "name": "Shqip",
    },
    "sr": Object {
      "key": "sr",
      "locale": "sr_RS",
      "localizedName": "Serbian",
      "name": "Српски",
    },
    "sv": Object {
      "key": "sv",
      "locale": "sv_SE",
      "localizedName": "Swedish",
      "name": "Svenska",
    },
    "sw": Object {
      "key": "sw",
      "locale": "sw_TZ",
      "localizedName": "Swahili",
      "name": "Swahili",
    },
    "ta": Object {
      "key": "ta",
      "locale": "ta_IN",
      "localizedName": "Tamil",
      "name": "தமிழ்",
    },
    "th": Object {
      "key": "th",
      "locale": "th_TH",
      "localizedName": "Thai",
      "name": "ภาษาไทย",
    },
    "tl": Object {
      "key": "tl",
      "locale": "tl_PH",
      "localizedName": "Filipino",
      "name": "Filipino",
    },
    "tr": Object {
      "key": "tr",
      "locale": "tr_TR",
      "localizedName": "Turkish",
      "name": "Türkçe",
    },
    "uk": Object {
      "key": "uk",
      "locale": "uk_UA",
      "localizedName": "Ukrainian",
      "name": "Українська",
    },
    "ur": Object {
      "direction": "rtl",
      "key": "ur",
      "locale": "ur_PK",
      "localizedName": "Urdu",
      "name": "اُردُو",
    },
    "vi": Object {
      "key": "vi",
      "locale": "vi_VN",
      "localizedName": "Vietnamese",
      "name": "Tiếng Việt",
    },
    "zh": Object {
      "key": "zh",
      "locale": "zh_CN",
      "localizedName": "Chinese",
      "name": "简体中文",
    },
    "zu": Object {
      "key": "zu",
      "locale": "zu_ZA",
      "localizedName": "Zulu",
      "name": "isiZulu",
    },
  },
  "deviceLanguage": Object {
    "key": "en",
    "locale": "en_US",
    "localizedName": "English",
    "name": "English",
  },
}
`;

exports[`connector/fallback connect send function accepts all comands command.mobile 1`] = `false`;

exports[`connector/fallback connect send function accepts all comands command.native 1`] = `false`;

exports[`connector/fallback connect send function accepts all comands command.nativeShare 1`] = `undefined`;

exports[`connector/fallback connect send function accepts all comands command.nativeUpload 1`] = `undefined`;

exports[`connector/fallback connect send function accepts all comands command.openLink 1`] = `undefined`;

exports[`connector/fallback connect send function accepts all comands command.prefContentLang 1`] = `"en_US"`;

exports[`connector/fallback connect send function accepts all comands command.version 1`] = `"3.4"`;
